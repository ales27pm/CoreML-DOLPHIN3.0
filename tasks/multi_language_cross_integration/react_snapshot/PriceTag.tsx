import React from "react";

type PriceTagProps = {
  readonly amount: number;
  readonly currency?: string;
  readonly locale?: string;
  readonly strikeThrough?: boolean;
};

const DEFAULT_LOCALE = "en-US";
const DEFAULT_CURRENCY = "USD";

export const PriceTag: React.FC<PriceTagProps> = ({
  amount,
  currency = DEFAULT_CURRENCY,
  locale = DEFAULT_LOCALE,
  strikeThrough = false,
}) => {
  const formatted = React.useMemo(
    () =>
      new Intl.NumberFormat(locale, {
        currency,
        style: "currency",
        currencyDisplay: "symbol",
      }).format(amount),
    [amount, currency, locale],
  );

  if (strikeThrough) {
    return (
      <span
        aria-label={`Original price ${formatted}`}
        style={{ textDecoration: "line-through" }}
      >
        {formatted}
      </span>
    );
  }

  return <span aria-label={`Current price ${formatted}`}>{formatted}</span>;
};

PriceTag.displayName = "PriceTag";

export default PriceTag;
