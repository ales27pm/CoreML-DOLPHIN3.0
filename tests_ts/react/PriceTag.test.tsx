import React from "react";
import renderer from "react-test-renderer";
import { describe, expect, it } from "vitest";

import { PriceTag } from "../../tasks/multi_language_cross_integration/react_snapshot/PriceTag";

describe("PriceTag", () => {
  it("renders formatted currency", () => {
    const tree = renderer
      .create(<PriceTag amount={123.45} currency="EUR" locale="de-DE" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("applies strike-through style when requested", () => {
    const tree = renderer
      .create(<PriceTag amount={99.99} strikeThrough />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
