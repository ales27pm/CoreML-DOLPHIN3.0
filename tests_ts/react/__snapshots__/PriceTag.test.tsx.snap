// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PriceTag > applies strike-through style when requested 1`] = `
<span
  aria-label="Original price $99.99"
  style={
    {
      "textDecoration": "line-through",
    }
  }
>
  $99.99
</span>
`;

exports[`PriceTag > renders formatted currency 1`] = `
<span
  aria-label="Current price 123,45 €"
>
  123,45 €
</span>
`;
